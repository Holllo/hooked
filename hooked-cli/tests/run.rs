use {assert_cmd::Command, insta::assert_snapshot, test_case::test_case};

#[test_case("pre-commit", "pre-commit.toml", true; "pre-commit")]
fn test_hooked_run(hook_type: &str, config: &str, expect_failure: bool) {
  let config = format!("tests/samples/{}", config);

  let mut command = Command::cargo_bin("hooked").unwrap();
  command
    .args(["run", hook_type, "--config", &config])
    .env("NO_COLOR", "1");

  let assert = if expect_failure {
    command.assert().failure().code(1)
  } else {
    command.assert().success()
  };

  let stderr = std::str::from_utf8(&assert.get_output().stderr).unwrap();
  let stdout = std::str::from_utf8(&assert.get_output().stdout).unwrap();
  assert_snapshot!(hook_type, format!("{}{}", stdout, stderr));
}
