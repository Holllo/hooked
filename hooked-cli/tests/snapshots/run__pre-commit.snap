---
source: hooked-cli/tests/run.rs
expression: "format!(\"{}{}\", stdout, stderr)"
---
Hooked: Running 3 pre-commit hooks.
	✓ Unnamed Hook
	⚠ Exit 1 Continue
on_failure=continue

	✗ Exit 1 Stop
Hey from stdout!
Hey from stderr!


